facechain/data_process/preprocessing.py

```diff
@@ -197,6 +250,64 @@ def get_mask_head(result):
     mask_head = np.expand_dims(mask_head, 2)
     return mask_head
 
+def compare_jpg_with_face_id(embedding_list):
+    embedding_array = np.vstack(embedding_list)
+    # average embedding 
+    pivot_feature   = np.mean(embedding_array, axis=0)
+    pivot_feature   = np.reshape(pivot_feature, [512, 1])
+    # get scores
+    scores = [np.dot(emb, pivot_feature)[0][0] for emb in embedding_list]
+    return scores
+
+def most_similar_faces_pick(imdir, face_detection, face_recognition, face_quality_func):
+    # ---------------------------calculate faceid best jpgs -------------------------- #
+    imlist = os.listdir(imdir)
+
+    face_id_scores  = []
+    quality_scores  = []
+    face_angles     = []
+    selected_paths  = []
+    for index, jpg in enumerate(tqdm(imlist)):
+        try:
+            if not jpg.lower().endswith(('.bmp', '.dib', '.png', '.jpg', '.jpeg', '.pbm', '.pgm', '.ppm', '.tif', '.tiff')):
+                continue
+            img_path = os.path.join(imdir, jpg)
+            image       = Image.open(img_path)
+            h, w, c     = np.shape(image)
+
+            retinaface_box, retinaface_keypoint, _ = call_face_crop(face_detection, image, 3, prefix="tmp")
+            retinaface_keypoint = np.reshape(retinaface_keypoint, [5, 2])
+            # caclculate angle
+            x = retinaface_keypoint[0,0] - retinaface_keypoint[1,0]
+            y = retinaface_keypoint[0,1] - retinaface_keypoint[1,1]
+            angle = 0 if x==0 else abs(math.atan(y/x)*180/math.pi)
+            angle = (90 - angle)/ 90 
+
+            # face width
+            face_width  = (retinaface_box[2] - retinaface_box[0]) / (3 - 1)
+            face_height = (retinaface_box[3] - retinaface_box[1]) / (3 - 1)
+            if face_width / w < 1/8 or face_height / h < 1/8:
+                continue
+
+            sub_image = image.crop(retinaface_box)
+
+            embedding   = np.array(face_recognition(sub_image)[OutputKeys.IMG_EMBEDDING])
+            score       = face_quality_func(sub_image)[OutputKeys.SCORES]
+            score       = 0 if score is None else score[0]
+
+            face_id_scores.append(embedding)
+            quality_scores.append(score)
+            face_angles.append(angle)
+            selected_paths.append(jpg)
+        except:
+            print(jpg, "is error skip")
```

Comment:
excpet e: 
skipping {jpg} due to processing error {e}

Reply:
done
fix to
except Exception as e:
    print(f'skipping {jpg} due to processing error {e}')